Article comparison metric

ansonmiu0214 opened the pull request:
Addresses cockatoos/sprint-board#18

Adds an ArticleComparison service which implements a `compare` method to return the number of edits to transform the recorded phrase to the ground truth phrase. The method is case-insensitive and only considers alphanumeric characters. The method returns a list of edits (a union of either insertion, deletion, replacement, or no-op) for another UI component to render the diff.

Unit tests included for the main use cases.

ansonmiu0214 committed:
Implement article comparison service

ansonmiu0214 committed:
Add unit tests for article comparison service

ansonmiu0214 committed:
Merge branch 'main' into article-comparison-metric

ansonmiu0214 committed:
Fix typos in test cases and make them clearer

ansonmiu0214 committed:
Fix whitespace

neshang commented:
The code change looks good! Should we move it to **app/services/article-comparison** as there might be more services in the future, might be quite messy if we put everything in the root src folder

ansonmiu0214 committed:
Move services to separate directory

ansonmiu0214 commented:
> The code change looks good! Should we move it to **app/services/article-comparison** as there might be more services in the future, might be quite messy if we put everything in the root src folder

Makes sense, addressed comments; please merge (once CI checks are done!)

Thanks :)
